### PR TITLE
feat: recurring tasks (daily/weekly/monthly) with auto-next occurrence

### PR DESCRIPTION
**Pull request by ethanzrd**
# feat: recurring tasks (daily/weekly/monthly) with auto-next occurrence

## Summary
This PR adds support for recurring tasks to the TodoApp. Users can now select daily, weekly, or monthly recurrence options when creating or editing tasks. When a recurring task is marked complete, the system automatically creates the next occurrence with the due date shifted by the chosen interval.

**Key Changes:**
- Extended `Task` interface with optional `recurrence` field supporting "daily", "weekly", "monthly"
- Added date manipulation utilities (`addDays`, `addWeeks`, `addMonths`) for shifting due dates
- Updated Add Task and Edit Task forms with recurrence selector dropdown
- Modified single task completion logic to generate next occurrence when recurring task is marked done
- Updated bulk task completion to handle multiple recurring tasks simultaneously
- Preserved all existing task properties (categories, colors, etc.) in next occurrences

## Review & Testing Checklist for Human
- [ ] **End-to-end flow testing**: Create a recurring task with a deadline, mark it complete, verify next occurrence appears with correctly shifted due date
- [ ] **All recurrence types**: Test daily (next day), weekly (next week), monthly (next month) - pay special attention to month-end dates (e.g., Jan 31 → Feb 28/29)
- [ ] **UI/UX verification**: Check that recurrence selectors look good and fit existing design patterns in both Add Task and Edit Task forms
- [ ] **Bulk operations**: Select multiple recurring tasks and mark them all complete simultaneously - verify each gets its next occurrence created
- [ ] **Edge cases**: Test tasks without deadlines (should only toggle done, no next occurrence), editing recurrence of existing tasks, month overflow scenarios

**Recommended Test Plan**: Create tasks with deadlines set to Jan 30, Feb 28, and Mar 31 with monthly recurrence, then mark them complete and verify the next occurrences have sensible dates.

---

### Diagram
```mermaid
%%{ init : { "theme" : "default" }}%%
graph TD
    UserTS["src/types/user.ts<br/>Task interface"]:::major-edit
    TimeUtils["src/utils/timeUtils.ts<br/>Date helpers"]:::major-edit
    AddTask["src/pages/AddTask.tsx<br/>Create form"]:::major-edit
    EditTask["src/components/tasks/EditTask.tsx<br/>Edit dialog"]:::major-edit
    TaskMenu["src/components/tasks/TaskMenu.tsx<br/>Single completion"]:::major-edit
    TasksList["src/components/tasks/TasksList.tsx<br/>Bulk completion"]:::major-edit
    
    UserContext["UserContext<br/>State management"]:::context

    UserTS --> AddTask
    UserTS --> EditTask
    UserTS --> TaskMenu
    UserTS --> TasksList
    TimeUtils --> TaskMenu
    TimeUtils --> TasksList
    AddTask --> UserContext
    EditTask --> UserContext
    TaskMenu --> UserContext
    TasksList --> UserContext

    subgraph Legend
        L1[Major Edit]:::major-edit
        L2[Minor Edit]:::minor-edit
        L3[Context/No Edit]:::context
    end

    classDef major-edit fill:#90EE90
    classDef minor-edit fill:#87CEEB
    classDef context fill:#FFFFFF
```

### Notes
- **Month overflow handling**: The `addMonths` function handles edge cases like Jan 31 → Feb 28/29 by setting to the last day of the target month
- **Backward compatibility**: All existing tasks remain unaffected as `recurrence` is optional
- **Performance**: Next occurrences are created immediately in memory, no async operations
- **State preservation**: Next occurrences inherit all properties from the original task except `id`, `done`, `date`, and `deadline`

**Session Info**: Implemented by Devin AI  
**Link to Devin run**: https://app.devin.ai/sessions/5fcbddd3b37549a6a55e5988905fe2db  
**Requested by**: Ethan Zerad (@ethanzrd)

### CODE REVIEW

File: src/components/tasks/EditTask.tsx [hunk 1]
```diff
@@ -241,6 +243,29 @@ export const EditTask = ({ open, task, onClose }: EditTaskProps) => {
             },
           }}
         />
+        <StyledInput
+          label="Recurrence"
+          name="recurrence"
+          select
+          value={editedTask?.recurrence || "none"}
+          onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
```
Comment (ethanzrd) [inline src/components/tasks/EditTask.tsx:251]: What does this line of code do?
